service/client.ts

```diff
@@ -23,6 +24,42 @@ import isGrpcError from "utils/isGrpcError";
 
 const URL = process.env.NEXT_PUBLIC_API_BASE_URL;
 
+async function http<T>(endpoint: string, configOverride: RequestInit): Promise<T> {
+  const url = path.join(URL, endpoint)
```

Comment:
```suggestion
  const url = new URL(endpoint, API_URL);
```
I believe `path` is a Node only API and so I don't think we should use it. Even if it works, it probably ends up bundling an unnecessary dependency. You can achieve the same with the URL class that's built into the web instead. I guess you'll probably have to rename `URL` to `API_URL` too as that'll likely be shadowing the actual URL class.